enhancements/sig-architecture/20-taint-toleration/README.md

```diff
@@ -0,0 +1,318 @@
+* `scheduleResult` should be updated by adding `requeue` and `delay`, which represent whether requeue need to be triggered in the controller and after how many seconds it would be triggerd. 
+* We also add a requeue process in the controller to evict the cluster after this `delay` time. `delay` time is the first expiration time of tolerated taints, and it would be computed as: toleration.TolerationSeconds - (current time - taint.TimeAdded), and the first expiration time is the minimum among all expiration times.
+* Update the plugin interface, which can return the `requeueAfter` time for each plugin.
+
+### Register repo
+
+#### Add new controller:
+
+Based on what we have in current Placement APIs, we add a new controller which can map the state of unhealthy and unreachable managed clusters to corresponding taints `open-clustet-management.io/unhealthy` and `open-cluster-management.io/unreachable`, so we can leverage the Taint-Toleration to support[ filtering unhealthy/not-reporting clusters (issue #48)](https://github.com/open-cluster-management-io/community/issues/48).
+
+### Examples
+
+#### 1. Cluster Maintaining
+
+Users (admins) can set a cluster to maintaining status, so workloads will be evicted and no workloads will be scheduled to the cluster. There's no need to set the toleration in placement.yaml under this senario. If toleration has to be set, make sure there won't be a matching key-value with the `maintaining` key-value in taint.
+
+cluster.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: cluster1
+  labels:
+    cluster.open-cluster-management.io/clusterset: clusterset1
+spec:
+  hubAcceptsClient: true
+  taints:
+    - key: maintaining
+      value: "true"
+```
+
+placement.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement1
+  namespace: default
+spec: ~
+```
+
+#### 2. Schedule on Certain Clusters
+
+Developers want to schedule some workloads to certain managed clusters.
+
+cluster.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: cluster1
+  labels:
+    cluster.open-cluster-management.io/clusterset: clusterset1
+spec:
+  hubAcceptsClient: true
+  taints:
+    - key: gpu
+      value: "true"
+```
+placement.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement1
+  namespace: default
+spec:
+  tolerations:
+    - key: gpu
+      value: "true"
+      operator: Equal
+```
+
+#### 3. Support Filtering Unhealthy/Unreachable Clusters
+
+When managed clusters' availability becomes `false`, the taint `unhealthy` would be automatically added to clusters. If managed clusters' availability becomes `unknown`, the taint `unreachable` would be added. Those unhealthy or unreachable clusters could be evicted immediatly or after `TolerationSeconds`.
```

Comment:
maybe open-cluster-management.io/unreachable and open-cluster-management.io/unhealthy